[GTK][WPE] Make size and mode construct only properties of Damage

carlosgcampos opened the pull request:
#### efc15638c8251a6e1f7b474609829cbf7451de6f
<pre>
[GTK][WPE] Make size and mode construct only properties of Damage
<a href="https://bugs.webkit.org/show_bug.cgi?id=290310">https://bugs.webkit.org/show_bug.cgi?id=290310</a>

Reviewed by Alejandro G. Castro.

And remove the resize and setMode methods. This ensures that all Damage
objects are created with a size and mode that can&apos;t change, except for
the case of making a damage full for which a new makeFull method is
added. This patch also remove the layout test
platform/glib/damage/storage-and-simplification-algorithm.html because
it assumes Damage is always 2048x1024 and it&apos;s now covered by the unit
tests.

* LayoutTests/platform/glib/damage/storage-and-simplification-algorithm-expected.txt: Removed.
* LayoutTests/platform/glib/damage/storage-and-simplification-algorithm.html: Removed.
* Source/WebCore/platform/graphics/Damage.h:
(WebCore::Damage::Damage):
(WebCore::Damage::makeFull):
(WebCore::Damage::initialize):
(WebCore::Damage::resize): Deleted.
(WebCore::Damage::setMode): Deleted.
* Source/WebCore/platform/graphics/texmap/TextureMapperLayer.cpp:
(WebCore::TextureMapperLayer::ensureDamageInLayerCoordinateSpace):
(WebCore::TextureMapperLayer::ensureDamageInGlobalCoordinateSpace):
(WebCore::TextureMapperLayer::damageWholeLayer):
* Source/WebCore/platform/graphics/texmap/coordinated/GraphicsLayerCoordinated.cpp:
(WebCore::GraphicsLayerCoordinated::updateDamage):
(WebCore::GraphicsLayerCoordinated::updateDirtyRegion):
(WebCore::GraphicsLayerCoordinated::commitLayerChanges):
* Source/WebCore/platform/graphics/texmap/coordinated/GraphicsLayerCoordinated.h:
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/ThreadedCompositor.cpp:
(WebKit::ThreadedCompositor::paintToCurrentGLContext):
* Source/WebKit/WebProcess/WebPage/dmabuf/AcceleratedSurfaceDMABuf.cpp:
(WebKit::AcceleratedSurfaceDMABuf::didRenderFrame):
(WebKit::AcceleratedSurfaceDMABuf::RenderTarget::didRenderFrame): Deleted.
* Source/WebKit/WebProcess/WebPage/dmabuf/AcceleratedSurfaceDMABuf.h:
* Tools/TestWebKitAPI/Tests/WebCore/glib/Damage.cpp:
(TestWebKitAPI::TEST(Damage, Basics)):
(TestWebKitAPI::TEST(Damage, Mode)):
(TestWebKitAPI::TEST(Damage, Move)):
(TestWebKitAPI::TEST(Damage, Resize)):
(TestWebKitAPI::TEST(Damage, AddRect)):
(TestWebKitAPI::TEST(Damage, AddDamage)):
(TestWebKitAPI::TEST(Damage, Unite)):

Canonical link: <a href="https://commits.webkit.org/292640@main">https://commits.webkit.org/292640@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0a6bedc52d9d8fc17551f48634192e0431188b8f

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/96550 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/16164 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/6262 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/101622 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/47070 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/98595 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/16460 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/24598 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/73596 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/30822 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/99553 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/12382 "Found 1 new test failure: fast/forms/ios/focus-input-via-button.html (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/87318 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/53932 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/12139 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/5104 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/46398 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/82247 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/5198 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/103646 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/23617 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/17221 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/82642 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/23868 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/83360 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/82017 "Found 3 new API test failures: /WebKitGTK/TestPrinting:/webkit/WebKitPrintOperation/print-errors, /WebKitGTK/TestPrinting:/webkit/WebKitPrintOperation/print, /WebKitGTK/TestPrinting:/webkit/WebKitPrintOperation/close-after-print (failure)") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/26670 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/4189 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/17091 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/15567 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/23580 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/28735 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/23239 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/26719 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/24980 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

spenap commented on Source/WebCore/platform/graphics/Damage.h:
```diff
@@ -46,11 +46,26 @@ class Damage {
+        , m_tileSize(s_tileSize, s_tileSize)
+    {
+        initialize();
     }
```
```suggestion
    explicit Damage(const FloatSize& size, Mode mode = Mode::Rectangles)
        : Damage(ceiledIntSize(LayoutSize(size)), mode)
    {
    }
```

we can do delegated constructors instead.

Scony commented on Source/WebCore/platform/graphics/texmap/coordinated/GraphicsLayerCoordinated.cpp:
```diff
@@ -892,29 +892,39 @@ void GraphicsLayerCoordinated::updateVisibleRect(const FloatRect& rect)
+                return { minDamageSize, minDamageSize };
+
+            if (m_size.width() > visibleSize.width() || m_size.height() > visibleSize.height())
+                return visibleSize;
```
IMO in case of e.g. 3000x512 layer and 2000x1000 viewport, the above should return 2000x512
similarly, IMO in case of e.g. 1000x256 layer and 2000x1000 viewport, the above should return 1000x512

so in other words, we should:
```
return {
min(max(m_size.width(), 512), visibleSize.width()),
min(max(m_size.height(), 512), visibleSize.height())
}
```

Scony reviewed (commented):
LGTM

alexgcastro reviewed (approved):
LGTM!

carlosgcampos committed:
[GTK][WPE] Make size and mode construct only properties of Damage
https://bugs.webkit.org/show_bug.cgi?id=290310

Reviewed by Alejandro G. Castro.

And remove the resize and setMode methods. This ensures that all Damage
objects are created with a size and mode that can't change, except for
the case of making a damage full for which a new makeFull method is
added. This patch also remove the layout test
platform/glib/damage/storage-and-simplification-algorithm.html because
it assumes Damage is always 2048x1024 and it's now covered by the unit
tests.

* LayoutTests/platform/glib/damage/storage-and-simplification-algorithm-expected.txt: Removed.
* LayoutTests/platform/glib/damage/storage-and-simplification-algorithm.html: Removed.
* Source/WebCore/platform/graphics/Damage.h:
(WebCore::Damage::Damage):
(WebCore::Damage::makeFull):
(WebCore::Damage::initialize):
(WebCore::Damage::resize): Deleted.
(WebCore::Damage::setMode): Deleted.
* Source/WebCore/platform/graphics/texmap/TextureMapperLayer.cpp:
(WebCore::TextureMapperLayer::ensureDamageInLayerCoordinateSpace):
(WebCore::TextureMapperLayer::ensureDamageInGlobalCoordinateSpace):
(WebCore::TextureMapperLayer::damageWholeLayer):
* Source/WebCore/platform/graphics/texmap/coordinated/GraphicsLayerCoordinated.cpp:
(WebCore::GraphicsLayerCoordinated::updateDamage):
(WebCore::GraphicsLayerCoordinated::updateDirtyRegion):
(WebCore::GraphicsLayerCoordinated::commitLayerChanges):
* Source/WebCore/platform/graphics/texmap/coordinated/GraphicsLayerCoordinated.h:
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/ThreadedCompositor.cpp:
(WebKit::ThreadedCompositor::paintToCurrentGLContext):
* Source/WebKit/WebProcess/WebPage/dmabuf/AcceleratedSurfaceDMABuf.cpp:
(WebKit::AcceleratedSurfaceDMABuf::didRenderFrame):
(WebKit::AcceleratedSurfaceDMABuf::RenderTarget::didRenderFrame): Deleted.
* Source/WebKit/WebProcess/WebPage/dmabuf/AcceleratedSurfaceDMABuf.h:
* Tools/TestWebKitAPI/Tests/WebCore/glib/Damage.cpp:
(TestWebKitAPI::TEST(Damage, Basics)):
(TestWebKitAPI::TEST(Damage, Mode)):
(TestWebKitAPI::TEST(Damage, Move)):
(TestWebKitAPI::TEST(Damage, Resize)):
(TestWebKitAPI::TEST(Damage, AddRect)):
(TestWebKitAPI::TEST(Damage, AddDamage)):
(TestWebKitAPI::TEST(Damage, Unite)):

Canonical link: https://commits.webkit.org/292640@main
